scheduler/src/cook/config.clj

```diff
@@ -330,6 +330,9 @@
                                (or (:framework-name mesos) "Cook")))
      :mesos-framework-id (fnk [[:config {mesos nil}]]
                            (:framework-id mesos))
+     :mesos-compute-cluster-name (fnk [[:config {mesos nil}]]
+                                   (or (:compute-cluster-name mesos) "default-compute-cluster-from-config-defaulting"))
```

Comment:
What's the use case for the default name? I really don't like the constant, can we just get rid of it?

Reply:
For legacy jobs in the database, I want to give them a compute cluster. We don't have to do this, but if we don't, then :instance/compute-cluster is an optional key under schema validation. 

And this means we have to keep the code path of having it and not having it alive forever. If we always have a cluster (even for historic jobs that 'didn't really have one', then the key can be made mandatory, and other code (including internal code) can assume it always exists, even if with (correct) filler data.

Reply:
I can see that, but then can we kill this config line and just require a mesos compute cluster to be defined for now? In the future if that's no longer the case, we'll have a more complicated scheme to backfill anyway.

Reply:
Where did we leave this? I thought we were going to make it required in config and crash if not defined.

Reply:
Bump

Reply:
We decided to backfill historic records with a compute cluster on read, that way we don't have to have dual codepaths for if there is or there isn't a compute cluster in every code that tries to use it (e.g., every place where we attempt to kill a job.)